torchdynamo/variables/builder.py

```diff
@@ -202,6 +202,13 @@ def _wrap(self, value):
                 value=value,
                 guards=make_guards(GuardBuilder.CONSTANT_MATCH),
             )
+        elif isinstance(value, frozenset):
```

Comment:
I think we also need something like:
```
and all(is_allowed(x) or ConstantVariable.is_literal(x) for x in value)
```

You could have something like:
`frozenset([tensor, dict])`

Then someone could mutate the contents of the frozenset (without actually mutating the containing set).

Reply:
sure. Do we need any extra guards, or is this additional check enough?

I'm asking because before currently guarding on something like a `torch.add` is not doable with the current code, so if this needs additional guards maybe we would need to extend the dictionary keys guards to support guarding by ID, currently they only support literals and guard by equality checks.

Reply:
I'm guessing if ` all(is_allowed(x) or ConstantVariable.is_literal(x) for x in value)` no guards are needed other than ID of the set, since we're enforcing the set members to be immutable, assuming the is_allowed check only passes for immutable things?